contracts/exchange/mixins/auction/EndemicReserveAuction.sol

```diff
@@ -54,83 +51,71 @@ abstract contract EndemicReserveAuction is
         bytes32 r,
         bytes32 s,
         ReserveAuction calldata auction,
-        ReserveAuction calldata bid
-    ) external onlySupportedERC20Payments(auction.paymentErc20TokenAddress) {
-        if (
-            auction.isBid ||
-            !bid.isBid ||
-            auction.nftContract != bid.nftContract ||
-            auction.tokenId != bid.tokenId ||
-            auction.paymentErc20TokenAddress != bid.paymentErc20TokenAddress ||
-            auction.signer == bid.signer
-        ) revert InvalidConfiguration();
-
-        _verifyApprovalSignature(v, r, s, auction, bid);
-        _verifySignature(auction);
-        _verifySignature(bid);
-
-        AuctionFees memory auctionFees = _calculateAuctionFees(auction, bid);
-
-        if (auction.price + auctionFees.takerCut > bid.price) {
+        ReserveAuction calldata bid,
+        AuctionInfo calldata info
+    ) external onlySupportedERC20Payments(info.paymentErc20TokenAddress) {
+        if (auction.isBid || !bid.isBid || auction.signer == bid.signer) {
+            revert InvalidConfiguration();
+        }
+
+        _verifyApprovalSignature(v, r, s, auction, bid, info);
+        _verifySignature(auction, info);
+        _verifySignature(bid, info);
+
+        uint256 bidPrice = (bid.price * MAX_FEE) /
+            (info.takerFeePercentage + MAX_FEE);
```

Comment:
Je li ovo nesto sta bi trebali sad izracunat off-chain?

Ja sam to zamislia da sve vec bude izracunato i potpisano off-chain

Reply:
Nez jesam li razumia pitanje. Isto je sve kao prije sta se tice price, samo sto se jos potpisuju parametri: makerFeePercentage, takerFeePercentage, royaltiesPercentage, royaltiesRecipient. Oni bi morali bit isti u sva 3 potpisa: seller, bidder i endemic approval. Stavia sam da se potpisuju samo postotci, a ne iznosi, jer npr. priliko kreiranja aukcije ne mozemo znat koji ce bit iznos bida s kojim ce se settlat. Ovako sam isto napravia i kod Dutch aukcije. Napisi ako imas neki prijedlog.

Reply:
Dobro je ovo. Inicijalno smo zamislili da se potpisu iznosi, ali postotak je isto dobar upravo iz razloga koje si spomenia